Detections/MultipleDataSources/UserAgentSearch_log4j.yaml

```diff
@@ -0,0 +1,104 @@
+id: 29283b22-a1c0-4d16-b0a9-3460b655a46a
+name: User agent search for log4j exploitation attempt
+description: |
+  'This query uses various log sources having user agent data to look for log4j CVE-2021-44228 exploitation attempt based on user agent pattern. Log4j is an open-source Apache logging library that is used in 
+   many Java-based applications. The regex and the string matching look for the most common attacks. This might not be comprehensive to detect every possible user agent variation.
+   Reference: https://msrc-blog.microsoft.com/2021/12/11/microsofts-response-to-cve-2021-44228-apache-log4j2/'
+severity: High
+requiredDataConnectors:
+  - connectorId: WAF
+    dataTypes:
+      - AzureDiagnostics
+  - connectorId: Office365
+    dataTypes:
+      - OfficeActivity
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - SigninLogs
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - AADNonInteractiveUserSignInLogs
+  - connectorId: AWS
+    dataTypes:
+      - AWSCloudTrail
+  - connectorId: AzureMonitor(IIS)
+    dataTypes:
+      - W3CIISLog
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1190
+tags:
+    - log4j
+    - log4shell
+    - CVE2021-44228
+    - Schema: ASimWebSession
+    - SchemaVersion: 0.2.1
+    - Schema: ASimNetworkSessions
+    - SchemaVersion: 0.2.1
+query: |
+  let UserAgentString = dynamic (["${jndi:ldap:/", "${jndi:rmi:/", "${jndi:ldaps:/", "${jndi:dns:/", "${jndi:iiop:/","${jndi:","${jndi:nds:/","${jndi:corba/"]);
+  (union isfuzzy=true
+  (OfficeActivity
+  | where UserAgent has_any (UserAgentString) or UserAgent matches regex '\\$\\{j\\$\\{::-\\}n\\$\\{::-\\}d\\$\\{::-\\}[a-zA-Z]'
```

Comment:
Move the regex out of each query into a let statement.

Reply:
Thanks, Shain.  Made the changes as suggested.